src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCompiler.vb

```diff
@@ -17,13 +15,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Private Shared s_responseFileName As String
         Private ReadOnly _responseFile As String
         Private ReadOnly _diagnosticFormatter As CommandLineDiagnosticFormatter
+        Private _additionalTextFiles As ImmutableArray(Of AdditionalTextFile)
 
         Protected Sub New(parser As VisualBasicCommandLineParser, responseFile As String, args As String(), clientDirectory As String, baseDirectory As String, sdkDirectory As String, additionalReferenceDirectories As String, analyzerLoader As IAnalyzerAssemblyLoader)
             MyBase.New(parser, responseFile, args, clientDirectory, baseDirectory, sdkDirectory, additionalReferenceDirectories, analyzerLoader)
 
-            _diagnosticFormatter = New CommandLineDiagnosticFormatter(baseDirectory)
+            _diagnosticFormatter = New CommandLineDiagnosticFormatter(baseDirectory, AddressOf GetAdditionalTextFiles)
+            _additionalTextFiles = ImmutableArray(Of AdditionalTextFile).Empty
         End Sub
 
+        Private Function GetAdditionalTextFiles() As ImmutableArray(Of AdditionalTextFile)
```

Comment:
Well, we cannot have an external file diagnostic until an analyzer is invoked, which in turn cannot happen until we have resolved external file arguments and wrapped them into AnalyzerOptions to pass them into the analyzer.
I'll add an assert to ensure this invariant is held.